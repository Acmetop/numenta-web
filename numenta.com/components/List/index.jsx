// Numenta company website source code. Copyright Â© 2016 Numenta.
// Full details in LICENSE.txt, or contact us at <http://numenta.com>.
// This program is free software: you can redistribute it and/or modify it under
// the terms of the GNU Affero General Public License as published by the Free
// Software Foundation, either version 3 of the License, or (at your option) any
// later version. This program is distributed in the hope that it will be
// useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero
// General Public License for more details. You should have received a copy of
// the GNU Affero General Public License along with this program. If not, see
// <https://www.gnu.org/licenses/agpl.html>.

import React from 'react'

import styles from './index.css'


const List = ({children, copy, marker}) => {
  const childrenWithProps = React.Children.map(children, (child) =>
    React.cloneElement(child, {copy, marker})
  )

  return (
    <ul className={styles.ul}>
      {childrenWithProps}
    </ul>
  )
}

List.propTypes = {
  children: React.PropTypes.node.isRequired,
  copy: React.PropTypes.bool,
  marker: React.PropTypes.oneOf(['circle', 'disc', 'icon', 'square']),
}

List.defaultProps = {
  copy: true,
}

export default List
