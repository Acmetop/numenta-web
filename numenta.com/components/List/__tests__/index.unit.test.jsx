import React from 'react'
import renderer from 'react-test-renderer'

import List from '../../List'


describe('<List /> React component unit test suite', () => {

  it('Renders correctly', () => {
    const component = renderer.create(
      <List />
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

})
