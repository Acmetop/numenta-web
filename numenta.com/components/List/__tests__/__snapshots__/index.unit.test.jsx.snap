exports[`<List /> React component unit test suite Renders correctly with circle marker 1`] = `
<ul
  className="ul">
  <li
    className="li copy markerCircle">
    List Item 1
  </li>
  <li
    className="li copy markerCircle">
    List Item 2
  </li>
  <li
    className="li copy markerCircle">
    List Item 3
  </li>
</ul>
`;

exports[`<List /> React component unit test suite Renders correctly with default props 1`] = `
<ul
  className="ul">
  <li
    className="li copy">
    List Item 1
  </li>
  <li
    className="li copy">
    List Item 2
  </li>
  <li
    className="li copy">
    List Item 3
  </li>
</ul>
`;

exports[`<List /> React component unit test suite Renders correctly with disc marker 1`] = `
<ul
  className="ul">
  <li
    className="li copy markerDisc">
    List Item 1
  </li>
  <li
    className="li copy markerDisc">
    List Item 2
  </li>
  <li
    className="li copy markerDisc">
    List Item 3
  </li>
</ul>
`;

exports[`<List /> React component unit test suite Renders correctly with text-copy prop 1`] = `
<ul
  className="ul">
  <li
    className="li copy">
    List Item 1
  </li>
  <li
    className="li copy">
    List Item 2
  </li>
  <li
    className="li copy">
    List Item 3
  </li>
</ul>
`;

exports[`<List /> React component unit test suite Renders correctly without text-copy prop 1`] = `
<ul
  className="ul">
  <li
    className="li">
    List Item 1
  </li>
  <li
    className="li">
    List Item 2
  </li>
  <li
    className="li">
    List Item 3
  </li>
</ul>
`;
